contactsheet.ts

```diff
@@ -0,0 +1,247 @@
+        height: 100%;
+        display: flex;
+        margin: 0;
+        justify-content: left;
+        background-color: #fafafa;
+      }
+
+      .table-container {
+        display: flex;
+        flex-flow: column nowrap;
+        background-color: white;
+        width: 600px;
+        margin: 0 auto;
+        border-radius: 4px;
+        border: 1px solid #DADADA;
+        box-shadow: 0px 1px 4px rgba(0, 0, 0, .08);
+      }
+
+      .table-row {
+        display: flex;
+        flex-flow: row nowrap;
+        width: 100%;
+        border-bottom: 1px solid #dadada;
+      }
+
+      .table-row:hover {
+        background-color: #F0F0F0;
+      }
+
+      .heading {
+        background-color: #ececec;
+        color: #3e3e3e;
+        font-weight: bold;
+        position: sticky;
+        top: 0;
+      }
+
+      .row-item {
+        display: flex;
+        flex: 1;
+        font-size: 14px;
+        padding: 0 0;
+        justify-content: center;
+        align-items: center;
+        transition: all 0.15s ease-in-out;
+      }
+
+      .row-sub-container {
+        display: flex;
+        flex-flow: column nowrap;
+        flex: 1;
+      }
+
+      .row-sub-container .row-item {
+        padding: 8px 0;
+        border-bottom: 1px solid #dadada;
+      }
+
+      .table-row:last-child,
+      .row-sub-container .row-item:last-child {
+        border-bottom: 0;
+      }
+    </style>
+  </head>
+  <body>
+  <div class="main-container">
+    <div class="table-container">
+      <div class="table-row heading">
+        <div class="row-item">MDI ID</div>
+        <div class="row-item shrink"></div>
+        <div class="row-item shrink"></div>
+        <div class="row-item">Font Awesome id</div>
+      </div>
+`;
+}
+
+function templateFooter() {
+  return `  </div>
+  </div>
+  </body>`;
+}
+
+async function build() {
+  if (!existsSync(dist)) {
+    await mkdir(dist);
+  }
+
+  await writeFile(path.resolve(dist, filename), templateHeader());
+  for (const mdiIconId in mapping.icons) {
+    const faID = mapping.icons[mdiIconId];
+    const faSVGPath = icons[mdiIconId];
+
+    let faIconName;
+    let mappingStateClass = 'not-mapped';
+    let faIconMappingInfoText = faID;
+
+    if (stats[mdiIconId] === MappingStatus.UNKNOWN_MAPPING) {
+      faIconMappingInfoText = `Unknown "${faID}"`;
+      mappingStateClass = 'wrong-mapping';
+    } else if (stats[mdiIconId] === MappingStatus.EMPTY_MAPPING) {
+      faIconMappingInfoText = 'Empty mapping';
+      mappingStateClass = 'wrong-mapping';
+    } else if (stats[mdiIconId] === MappingStatus.REMAPPED) {
+      mappingStateClass = 'mapped';
+      faIconName = faID;
+    }
+
+    const mdiIconData = getSvgDataMDI(mdiIconId);
+    const mdiSVG = renderTemplate(
+      mdiIconData.name,
+      mdiIconData.svgPathData,
+      viewBoxMdi,
+    );
+
+    const viewBoxDefault = {
+      width: 24,
+      height: 24,
+    };
+
+    const viewBox = viewBoxes[mdiIconId] || viewBoxDefault;
```

Comment:
Why's there a default, all icons should already have a value (the same value) in the mapping, no?

Reply:
yes... ..initially I hadn't set mdi icons viewBox... ...this line can be changed